src/app/features/admin/admin.routing.module.ts

```diff
@@ -152,6 +141,27 @@ const routes: Routes = [
       },
     ],
   },
+  {
+    path: 'emails',
+    children: [
+      {
+        path: '',
+        component: EmailsComponent,
+        data: { title: 'Emails' },
+
+        children: [
+          {
+            path: 'targeted-emails',
+            component: TargetedEmailsComponent,
+            data: { title: 'Targeted Emails' },
+            resolve: {
+              emailTemplates: EmailTemplateResolver,
+            },
+          },
+        ],
+      },
+    ],
+  },
```

Comment:
```suggestion
  {
    path: 'emails',
    children: [
      {
        path: '',
        component: EmailsComponent,
        data: { title: 'Emails' },
      }  
      {
        path: 'targeted-emails',
        component: TargetedEmailsComponent,
        data: { title: 'Targeted Emails' },
        resolve: {
          emailTemplates: EmailTemplateResolver,
        },
      },
    ],
  },
```
I think this could be simplified so that there's only one `children` array

Reply:
Thanks, but for this component it won't work like that I have tried that.